packages/rpc-methods/src/restricted/manageAccounts.test.ts

```diff
@@ -0,0 +1,198 @@
+import { SubjectType, PermissionType } from '@metamask/permission-controller';
+import { MOCK_SNAP_ID } from '@metamask/snaps-utils/test-utils';
+
+import {
+  methodName,
+  manageAccountsBuilder,
+  manageAccountsImplementation,
+  specificationBuilder,
+  validateParams,
+} from './manageAccounts';
+
+// To Do:
+// Move the class SnapKeyring to it's own module
+// add mock the method in this test instead of the entire class
+class SnapKeyringMock {
+  static type = 'Snap Keyring';
+
+  accounts: string[] = [];
+
+  handleKeyringSnapMessage = async (
+    _origin: string,
+    _params: any,
+    // eslint-disable-next-line @typescript-eslint/ban-types
+    _saveSnapKeyring: Function,
```

Comment:
Can we specify an actual function type here?
```suggestion
    _saveSnapKeyring: () => void,
```